web/client-api/src/main/java/io/deephaven/web/DeephavenJsApiLinker.java

```diff
@@ -54,23 +54,41 @@ public ArtifactSet link(TreeLogger logger, LinkerContext context, ArtifactSet ar
 
         // get the generated javascript
         String[] javaScript = result.getJavaScript();
-        out.print("self.dh = {}");
-        out.newline();
-        out.print("var $wnd = self, $doc, $entry, $moduleName, $moduleBase;");
-        out.newline();
-        out.print("var $gwt_version = \"" + About.getGwtVersionNum() + "\";");
-        out.newlineOpt();
-        out.print(javaScript[0]);
-        out.newline();
-
-        out.print("gwtOnLoad(null,'" + context.getModuleName() + "',null);");
-        out.newline();
-        out.print("})();");
-        out.newline();
+
+        StringBuffer buffer = readFileToStringBuffer(getSelectionScriptTemplate(), logger);
+        replaceAll(buffer, "__GWT_VERSION__", About.getGwtVersionNum());
+        replaceAll(buffer, "__JAVASCRIPT_RESULT__", javaScript[0]);
+        replaceAll(buffer, "__MODULE_NAME__", context.getModuleName());
+
+        out.print(buffer.toString());
 
         toReturn.add(emitString(logger, out.toString(), "dh-core.js"));
 
         return toReturn;
     }
 
+    protected StringBuffer readFileToStringBuffer(String filename,
+            TreeLogger logger) throws UnableToCompleteException {
+        StringBuffer buffer;
+        try {
+            buffer = new StringBuffer(Utility.getFileFromClassPath(filename));
+        } catch (IOException e) {
+            logger.log(TreeLogger.ERROR, "Unable to read file: " + filename, e);
+            throw new UnableToCompleteException();
+        }
+        return buffer;
+    }
+
+    protected String getSelectionScriptTemplate() {
+        return "io/deephaven/web/DeephavenJsApiLinkerTemplate.js";
+    }
+
+    protected static void replaceAll(StringBuffer buf, String search,
+            String replace) {
+        int len = search.length();
+        for (int pos = buf.indexOf(search); pos >= 0; pos = buf.indexOf(search,
+                pos + 1)) {
```

Comment:
I think this should be `pos + replace.length()` for correctness... if for example you are passing in a buf with `HELLOHELLO`, search `HELLO` and replace with empty string ``, you will only get rid of the first `HELLO`
By doing `+ replace.length()` you're handling that case and also skipping some chars that can't match.